app/controllers/post_favorites_controller.rb

```diff
@@ -1,24 +1,15 @@
-require_relative '../logical/danbooru/paginator/elasticsearch_extensions'
 class PostFavoritesController < ApplicationController
   before_action :member_only
   respond_to :html
 
   def index
     @post = Post.find(params[:post_id])
-    # order by most recent favorites first
-    all_users = @post.favorited_users_by_id.reverse
-    page = [params[:page].to_i, 1].max
-
-    paginated_users = all_users[(page - 1) * page_size, (page - 1) * page_size + page_size]
-    users = User.includes(:user_status).find(paginated_users).reject(&:hide_favorites?)
-
-    options = {mode: :numbered, per_page: page_size, total: all_users.length, current_page: page}
-    @users = ::Danbooru::Paginator::PaginatedArray.new(users, options)
-  end
-
-  private
-
-  def page_size
-    75
+    query = User.includes(:user_status).joins("INNER JOIN favorites ON favorites.user_id = users.id")
+    unless CurrentUser.is_admin?
+      query = query.where("bit_prefs & :value != :value", {value: 2**User::BOOLEAN_ATTRIBUTES.find_index("enable_privacy_mode")}).or(query.where(favorites: {user_id: CurrentUser.id}))
+    end
+    query = query.where(favorites: {post_id: @post.id})
+    query = query.order(favorites: :desc)
```

Comment:
This sort order doesn't do what you expect it does. This is sorting by "favorites".* which is all columns in the favorites table. It would be better to sort these by "users".name so that it's consistent. I don't see the utility in showing users with the most favorites first, it seems to incentivize people to make more favorites even though there is an explicit cap on favorite count.

Reply:
I thought it just orders by favorites.id which would show the most recent ones first. But sorting by name is fine too, I don' really prefer one over the otger

Reply:
I added explicit ordering by favorites.id which should result in the oldest favorites being show first. That way you can't cheat your way into the first page by refaving. But if I should just order by user.name do say so.

Reply:
I think ordering by name offers the least gamification of this parameter. Doing it by favorite id has some prestige around getting there first. By name should provide a reasonable compromise between being predictable, and not rewarding specific behaviors.

Reply:
I guess nothing would prevent someone from just trying to fav everything the moment it's posted (well, except the favlimit). Sorting by most recent would probably have been for novelty only anyways